api/tests/organizational-entities/integration/domain/usecases/archive-organization.usecase.test.js

```diff
@@ -1,26 +1,47 @@
 import { usecases } from '../../../../../src/organizational-entities/domain/usecases/index.js';
-import { databaseBuilder, expect, sinon } from '../../../../test-helper.js';
+import { NotFoundError } from '../../../../../src/shared/domain/errors.js';
+import { catchErr, databaseBuilder, expect, sinon } from '../../../../test-helper.js';
 
 describe('Integration | Organizational Entities | Domain | UseCase | archive-organization', function () {
-  it('archives the organization', async function () {
-    // given
-    const now = new Date('2022-02-22');
-    const clock = sinon.useFakeTimers({ now, toFake: ['Date'] });
-    const superAdminUser = databaseBuilder.factory.buildUser();
-    const organization = databaseBuilder.factory.buildOrganization();
-    await databaseBuilder.commit();
+  context('when the organization does exist', function () {
+    it('archives the organization', async function () {
+      // given
+      const now = new Date('2022-02-22');
+      const clock = sinon.useFakeTimers({ now, toFake: ['Date'] });
+      const superAdminUser = databaseBuilder.factory.buildUser();
+      const organization = databaseBuilder.factory.buildOrganization();
+      await databaseBuilder.commit();
 
-    // when
-    const archivedOrganization = await usecases.archiveOrganization({
-      organizationId: organization.id,
-      userId: superAdminUser.id,
+      // when
+      const archivedOrganization = await usecases.archiveOrganization({
+        organizationId: organization.id,
+        userId: superAdminUser.id,
+      });
+
+      // then
+      expect(archivedOrganization.archivedAt).to.deep.equal(now);
+      expect(archivedOrganization.archivistFirstName).to.deep.equal(superAdminUser.firstName);
+      expect(archivedOrganization.archivistLastName).to.deep.equal(superAdminUser.lastName);
+
+      clock.restore();
     });
+  });
 
-    // then
-    expect(archivedOrganization.archivedAt).to.deep.equal(now);
-    expect(archivedOrganization.archivistFirstName).to.deep.equal(superAdminUser.firstName);
-    expect(archivedOrganization.archivistLastName).to.deep.equal(superAdminUser.lastName);
+  context('when the organization does not exist', function () {
+    it('throws an error', async function () {
+      // given
+      const nonExistingOrganizationId = 123456;
+      const superAdminUser = databaseBuilder.factory.buildUser();
+      await databaseBuilder.commit();
 
-    clock.restore();
+      // when
+      const error = await catchErr(usecases.archiveOrganization)({
+        organizationId: nonExistingOrganizationId,
+        userId: superAdminUser.id,
+      });
+
+      // then
+      expect(error).to.be.instanceOf(NotFoundError);
```

Comment:
https://github.com/chaijs/chai-as-promised est plus adapté que `catchErr` pour tester les erreurs renvoyées par les promises :

```suggestion   
      // when & then
     await expect(usecases.archiveOrganization({
        organizationId: nonExistingOrganizationId,
        userId: superAdminUser.id,
      })).to.be.rejectedWith(NotFoundError);
```

Reply:
catchErr est un helper asynchrone maison utilisé un peu partout justement pour gérer les erreurs renvoyées par des promessses.

```
function catchErr(promiseFn, ctx) {
  return async (...args) => {
    try {
      await promiseFn.call(ctx, ...args);
    } catch (err) {
      return err;
    }
    throw new Error('Expected an error, but none was thrown.');
  };
} 
```

Et je ne suis vraiment pas fan de combiner when/then. Là les deux sont bien séparés, l'erreur est stockées dans une variable explicite, et c'est beaucoup plus lisible.